src/codegen/type/integer_type.cpp

```diff
@@ -374,6 +376,31 @@ struct Modulo : public TypeSystem::BinaryOperator {
   }
 };
 
+// DateTrunc
+struct DateTrunc : public TypeSystem::BinaryOperator {
```

Comment:
Add a `TODO` to move this to the `Timestamp` type once the function lookup logic is corrected.

Reply:
Do we have a plan on correcting the function lookup logic? Just curious.